internal/controllers/machine/machine_controller_status_test.go

```diff
@@ -0,0 +1,1533 @@
+			name:    "NodeReady missing from node",
+			machine: defaultMachine.DeepCopy(),
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{Type: corev1.NodeMemoryPressure, Status: corev1.ConditionFalse},
+						{Type: corev1.NodeDiskPressure, Status: corev1.ConditionFalse},
+						{Type: corev1.NodePIDPressure, Status: corev1.ConditionFalse},
+					},
+				},
+			},
+			expectConditions: []metav1.Condition{
+				{
+					Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  clusterv1.MachineNodeConditionNotYetReportedV1Beta2Reason,
+					Message: "Node Ready: condition not yet reported",
+				},
+				{
+					Type:   clusterv1.MachineNodeReadyV1Beta2Condition,
+					Status: metav1.ConditionUnknown,
+					Reason: clusterv1.MachineNodeConditionNotYetReportedV1Beta2Reason,
+				},
+			},
+		},
+		{
+			name: "node not found while machine is deleting",
+			machine: func() *clusterv1.Machine {
+				m := defaultMachine.DeepCopy()
+				m.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
+				return m
+			}(),
+			node: nil,
+			expectConditions: []metav1.Condition{
+				{
+					Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  clusterv1.MachineNodeDeletedV1Beta2Reason,
+					Message: "Node has been deleted",
+				},
+				{
+					Type:    clusterv1.MachineNodeReadyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  clusterv1.MachineNodeDeletedV1Beta2Reason,
+					Message: "Node has been deleted",
+				},
+			},
+		},
+		{
+			name: "node missing while machine is still running",
+			machine: func() *clusterv1.Machine {
+				m := defaultMachine.DeepCopy()
+				m.Status.NodeRef = &corev1.ObjectReference{
+					Name: "test-node-1",
+				}
+				return m
+			}(),
+			node: nil,
+			expectConditions: []metav1.Condition{
+				{
+					Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  clusterv1.MachineNodeDeletedV1Beta2Reason,
+					Message: "Node test-node-1 has been deleted while the machine still exist",
+				},
+				{
+					Type:    clusterv1.MachineNodeReadyV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  clusterv1.MachineNodeDeletedV1Beta2Reason,
+					Message: "Node test-node-1 has been deleted while the machine still exist",
+				},
+			},
+		},
+		{
+			name: "machine with ProviderID set, Node still missing",
+			machine: func() *clusterv1.Machine {
+				m := defaultMachine.DeepCopy()
+				m.Spec.ProviderID = ptr.To("foo://test-node-1")
+				return m
+			}(),
+			node: nil,
+			expectConditions: []metav1.Condition{
+				{
+					Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  clusterv1.MachineNodeNotFoundV1Beta2Reason,
+					Message: "Waiting for a node with Provider ID foo://test-node-1 to exist",
```

Comment:
```suggestion
					Message: "Waiting for a Node with Provider ID foo://test-node-1 to exist",
```
Let's try to use upper case for kind's in messages consistently